oshi-core/src/main/java/oshi/util/ParseUtil.java

```diff
@@ -489,4 +493,31 @@ public static int getNthIntValue(String line, int n) {
         }
         return 0;
     }
+
+    /**
+     * Removes all sub strings from the string
+     * @param original source String to remove from
+     * @param toRemove the sub string to be removed
+     * @return
+     */
+    public static String removeMatchingString(final String original, final String toRemove) {

```

Comment:
Well, the original code (both Apache and JDK) has two extra params I've removed to simplify the code: an int to specify the amount of times the match is searched for, and a boolean to set if you wanted case to be ignored.

We can add the boolean, move the method to private access, and then give two public methods, for removeAllMatchingStrings and removeFirstMatchingString, to keep from duplicating code.

Otherwise, we could implement:
```
String removeFirstMatchingString(final String original, String toRemove) {
    if (original == null || original.isEmpty() || toRemove == null || toRemove.isEmpty())
        return original;

    int matchIndex = original.indexOf(toRemove, 0);
    if (matchIndex == -1)
        return original;
    
    return original.substring(0, matchIndex) + original.substring(matchIndex + toRemove.length());
}
```

Anyways, I just realized there's a small error in my previous code. Replace
```
    ...
    StringBuilder buffer = new StringBuilder(original.length());
    int currIndex = 0;
    do {
        buffer.append(original.substring(currIndex, matchIndex));
        currIndex = matchIndex;
        matchIndex = original.indexOf(toRemove, currIndex);
    } while (matchIndex != -1);
    ...
```
with
```
    ...
    StringBuilder buffer = new StringBuilder(original.length());
    int currIndex = 0;
    int removedLength = toRemove.length();
    do {
        buffer.append(original.substring(currIndex, matchIndex));
        currIndex = matchIndex + removedLength;
        matchIndex = original.indexOf(toRemove, currIndex);
    } while (matchIndex != -1);
    ...
```
Or it won't step forward correctly